lib/custom/src/MShop/Service/Provider/Payment/Datatrans.php

```diff
@@ -108,4 +113,35 @@ protected function getXmlProvider()
 
 		return $provider;
 	}
+
+
+	/**
+	 * Queries for status updates for the given order compare with the responseCode
+	 *
+	 * @param \Aimeos\MShop\Order\Item\Iface $order
+	 */
+	public function query( \Aimeos\MShop\Order\Item\Iface $order )
+	{
+		$base = $this->getOrderBase( $order->getBaseId() );
```

Comment:
You should add `\Aimeos\MShop\Order\Item\Base\Base::PARTS_NONE` to speed up processing because you only need the base data